java/java.source.base/test/unit/src/org/netbeans/modules/java/source/indexing/VanillaCompileWorkerTest.java

```diff
@@ -685,6 +685,7 @@ public void testPreserveValidMethods1() throws Exception {
                 "        System.err.println(\"Hello, world!\");\n" +
                 "    }\n" +
                 "}");
+        file2Fixed.replaceAll((k, v) -> v.replaceAll("\r\n", "\n"));
         assertEquals(expected, file2Fixed);
```

Comment:
I am surprised that this is needed since golden files should be already covered by https://github.com/apache/netbeans/blob/master/.gitattributes (since they typically are `.java` files)

you did also mention
> the latter may also be addressed via git config --local core.autocrlf false;

right, but local git config adjustments should not be needed in our case since the nb repo should know the requirements already.

Reply:
Actually my mention on the `git config` adjustments pertains to the other file *java/hints/infrastructure/TreeRuleTestBase.java*, not this file *java/source/indexing/VanillaCompileWorkerTest.java*. 

1. In the latter, i.e. *VanillaCompileWorkerTest*:
    1. Only the ***actual*** output in `file2Fixed` is being processed to convert `\r\n` &rarr; `\n`.
    2. The  `expected` string is **not** changed.
    3. The possible reason why the generated output has `\r\n` in it is that on Windows, the module generates code with CRLF by default.
2. Regarding `git config`:
    1. Thank you for including the `.gitattributes` in the discussion.
    2. I checked it now and found that the *goldenfiles* in this package are named with the extension `.pass`, in addition to `.java`. 
    3. That is why the `git config` command was noted as a workaround alternative.
    4. I don't know if *.pass* files in other locations have different CRLF requirements.
    5. Please let me know if I should go ahead and include the following change to the *.gitattributes* file to set as binary (instead of `text eol=lf`):  
        - `*.pass             binary`
    6. However, there may be other places where *TreeRuleTestBase* could be extended in the future and the goldenfiles end up being named differently.
    7. Finally, if the SCM is changed, or a zip of the source is used, then would the code change in *TreeRuleTestBase.java* not still be needed?

Please let me know what you think. Thank you very much.

Reply:
> I checked it now and found that the goldenfiles in this package are named with the extension .pass, in addition to .java.

this might be it indeed. Localized `.pass` files could be mixed with portable Strings which causes trouble during comparison.

Unfortunately `.pass` files are used across the repo for other languages too and we might not have enough coverage on windows to tweak the repo settings for all of them. This is probably something we should do anyway but it doesn't have to be in this PR.

Overall I am ok with the proposed changes here - it just looked a bit suspicious to me that it is needed.

Reply:
ah, we already use more than one file!
https://github.com/apache/netbeans/blob/master/php/.gitattributes

and this line would confirm our suspicion:
https://github.com/apache/netbeans/blob/059bba1752779436b07acdabc64cb19c518672d5/php/.gitattributes#L5

so we could try the same here and reduce the scope to the java cluster (and check if all tests pass), or lower to only the modules which need it. Since absolute paths seem to work, we could simply add it to the root file without creating yet another file - maybe easier to maintain.

> Please let me know if I should go ahead and include the following change to the .gitattributes file to set as binary (instead of text eol=lf): 

don't use `binary` since reading the doc it sounds like it would affect more than just the separators, lets keep `text` but lock it to `eol=lf` like the `php` cluster does. I am curious if this already resolves the problem without this PR applied.

Reply:
Thank you very much. 

I have tested and pushed the commit with the *.gitattributes* set to mark `eol=lf` only for all files in the *goldenfiles* directory of the *java/java.hints* module unit-tests.
- This takes care of not requiring modification of the golden files during test assertions.

The patch still requires the changes to the actual output `System.lineSeparator()` is not LF:
1. *VanillaCompileWorkerTest*: replace `System.lineSeparator()` to `\n`
2. *TreeRuleTestBase*: fixed regex `[ \t\n]+` &rarr; `\\s+`